apps/cowswap-frontend/src/modules/swap/helpers/getAmountsForSignature.test.ts

```diff
@@ -0,0 +1,143 @@
+import { USDC_MAINNET, WETH_MAINNET } from '@cowprotocol/common-const'
+import { OrderKind } from '@cowprotocol/cow-sdk'
+import { Currency, CurrencyAmount, Percent, Price, TradeType } from '@uniswap/sdk-core'
+
+import TradeGp from 'legacy/state/swap/TradeGp'
+
+import { getAmountsForSignature } from './getAmountsForSignature'
+
+interface TradeParams {
+  tradeType: TradeType
+  inputAmount: CurrencyAmount<Currency>
+  outputAmount: CurrencyAmount<Currency>
+  feeAmount: CurrencyAmount<Currency>
+}
+
+function getTrade(params: TradeParams): TradeGp {
+  const { tradeType, inputAmount, outputAmount, feeAmount } = params
+
+  const isExactInput = tradeType === TradeType.EXACT_INPUT
+
+  const executionPrice = new Price({
+    baseAmount: inputAmount,
+    quoteAmount: outputAmount,
+  })
+
+  return new TradeGp({
+    inputAmount,
+    inputAmountWithFee: isExactInput ? inputAmount.subtract(feeAmount) : inputAmount.add(feeAmount),
+    inputAmountWithoutFee: inputAmount,
+    outputAmount: isExactInput ? outputAmount.subtract(executionPrice.quote(feeAmount)) : outputAmount,
+    outputAmountWithoutFee: outputAmount,
+    fee: { feeAsCurrency: feeAmount, amount: feeAmount.toFixed() },
+    executionPrice,
+    tradeType,
+    quoteId: 1,
+  })
+}
+
+describe('getAmountsForSignature()', () => {
+  // 5%
+  const allowedSlippage = new Percent(5, 100)
+  // 3012 USDC
+  const inputAmount = CurrencyAmount.fromRawAmount(USDC_MAINNET, 3012 * 10 ** USDC_MAINNET.decimals)
+  // 8 USDC
+  const feeAmount = CurrencyAmount.fromRawAmount(USDC_MAINNET, 8 * 10 ** USDC_MAINNET.decimals)
+  // 2 WETH
+  const outputAmount = CurrencyAmount.fromRawAmount(WETH_MAINNET, 2 * 10 ** WETH_MAINNET.decimals)
+
+  describe('With fee > 0', () => {
+    // Fee=zero is disabled
+    const featureFlags = { swapZeroFee: false }
+
+    /**
+     * Sell amount is just what user entered minus fee
+     * Output amount is calculated as: (outputAmount - fee) - slippage
+     */
+    it('Sell order', () => {
+      const trade = getTrade({ tradeType: TradeType.EXACT_INPUT, inputAmount, outputAmount, feeAmount })
+
+      const result = getAmountsForSignature({
+        featureFlags,
+        allowedSlippage,
+        trade: trade,
+        kind: OrderKind.SELL,
+      })
+
+      // Just subtracted fee from inputAmount, 3012 - 8 = 3004
+      expect(result.inputAmount.toFixed()).toEqual('3004.000000')
+
+      // Fee in WETH = 0.005312084993359893
+      // outputAmount = (2 - 0.005312) - 5% = 1.89468
```

Comment:
The result from this comment matches the 0-fee test.
The exact result for this one is different, as seen in the test itself on line `72`.

```suggestion
      // outputAmount = (2 - 0.005312) - 5% = 1.89495
```